inject-java/src/test/groovy/io/micronaut/inject/configproperties/eachbeanparameter/MyFactory.java

```diff
@@ -0,0 +1,24 @@
+package io.micronaut.inject.configproperties.eachbeanparameter;
+
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Parameter;
+
+@Factory
+public class MyFactory {
+
+    @EachBean(AbstractDataSource.class)
+    MyHelper buildHelper() {
+        return new MyHelper();
+    }
+
+    @Context // The context should load without properties and properly fill the parameter from DefaultDataSource
```

Comment:
test passes whether `@Context` is defined or not. Should we remove it?

Reply:
We should keep, it's was causing the problem

Reply:
With your code changes, it seems unnecessary but without the changes to  `BeanDefinitionDelegate` only one bean of type `MyBean` was registered.